ui-v2/src/components/ui/flow-run-acitivity-bar-graph/flow-run-activity-bar-graph.test.tsx

```diff
@@ -0,0 +1,152 @@
+import { render, screen, fireEvent, within } from "@testing-library/react";
+import { FlowRunActivityBarChart } from "./index";
+import { RouterContextProvider } from "@tanstack/react-router";
+import {
+	afterEach,
+	beforeEach,
+	vi,
+	describe,
+	it,
+	expect,
+	beforeAll,
+} from "vitest";
+import { router } from "@/router";
+
+const mockFlowRun = {
+	id: "test-flow-run-1",
+	name: "Test Flow Run",
+	state_type: "COMPLETED",
+	state: {
+		type: "COMPLETED",
+		name: "Completed",
+	},
+	start_time: "2024-01-01T00:00:00.000Z",
+	total_run_time: 3600,
+	tags: ["test-tag"],
+	deployment: {
+		id: "test-deployment-1",
+		name: "Test Deployment",
+	},
+	flow: {
+		id: "test-flow-1",
+		name: "Test Flow",
+	},
+};
+
+const mockEnrichedFlowRuns = [mockFlowRun];
+
+beforeAll(() => {
+	class ResizeObserverMock {
+		observe() {}
+		unobserve() {}
+		disconnect() {}
+	}
+
+	global.ResizeObserver = ResizeObserverMock;
```

Comment:
Is there a way to move this to tests/setup?

I'm also hitting a problem with this with dialogs and needed to copy/paste the same logic

Reply:
Yeah, I think if you move the entire `beforeAll` hook into `ui-v2/tests/setup.ts`, then all tests should get this mock by default. If specific tests need a different mock then they should be able to override it.